src/MeetingsSDKAdapter.js

```diff
@@ -273,19 +279,24 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    * @param {string} ID ID of the meeting to mute video
    * @memberof MeetingsSDKAdapter
    */
-  muteVideoMeeting(ID) {
+  async handleLocalVideo(ID) {
     const sdkMeeting = this.fetchMeeting(ID);
+    const videoEnabled = this.meetings[ID].localVideo.getVideoTracks()[0].enabled;
+    let eventName;
 
-    sdkMeeting
-      .muteVideo()
-      .then(() => {
-        // Due to SDK limitation around promises, we need to emit a custom event for video mute action
-        sdkMeeting.emit(EVENT_MEDIA_LOCAL_MUTE, {control: MUTE_VIDEO_CONTROL});
-      })
-      .catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(`Unable to mute video for meeting "${ID}"`, error);
-      });
+    try {
+      if (videoEnabled) {
+        await sdkMeeting.muteVideo();
+      } else {
+        await sdkMeeting.unmuteVideo();
+      }
+
+      // Due to SDK limitation around promises, we need to emit a custom event for video mute action
+      sdkMeeting.emit(eventName, {control: VIDEO_CONTROL, state: !videoEnabled});
```

Comment:
Similar comments as in the audio handling method:
* Where is `eventName` set?
* should you re-assign `videoEnabled` after muting/unmuting so we send the current state of the SDK/server track?